crates/ruma-identifiers/src/macros.rs

```diff
@@ -289,3 +289,175 @@ macro_rules! opaque_identifier {
         partial_eq_string!(Box<$id>);
     };
 }
+
+macro_rules! opaque_identifier_validated {
+    (
+        $( #[doc = $docs:literal] )*
+        $vis:vis type $id:ident;
+        $val_id:ident;
```

Comment:
I would expect `val` to abbreviate `value`, not `validate`. Maybe rename this to `$validate`, or `$validate_id`.